File: Source/DriveWorks.Sdk.Examples.VbDotNet/MyGenerationTaskSpecificationCondition.vb
Imports DriveWorks.Components.Tasks
Imports DriveWorks.Specification

''' <summary>
''' A Generation Task Specification Condition which determines whether the value entered is even.
''' </summary>
''' <remarks><see cref="ComponentTaskReleaseCondition"/> are evaluated at the time of releasing component-sets.</remarks>
<ComponentTaskCondition("Is Even",
                        "Passes if the specified value is even.",
                        "embedded://DriveWorks.Sdk.Examples.VbDotNet.Puzzle-16x16.png",
                        "SDK-Starter-Examples Plugin")>
Public Class MyGenerationTaskSpecificationCondition
    Inherits ComponentTaskReleaseCondition

    Private ReadOnly mValue As FlowProperty(Of Double) = Me.Properties.RegisterDoubleProperty("Value",
                                                                                              "The value to check for must be an even number for the condition to pass.")

    Protected Overrides Function Evaluate(specificationContext As SpecificationContext) As Boolean

        ' If a number is evenly divisible by 2 with no remainder, then it is even.
        Dim isEven = (mValue.Value Mod 2) = 0

        Return isEven

    End Function
End Class

File: Source/DriveWorks.Sdk.Examples.CSharp/MyGenerationTaskSpecificationCondition.cs
using DriveWorks.Components.Tasks;
using DriveWorks.Specification;

namespace DriveWorks.Sdk.Examples.CSharp
{
    /// <summary>
    /// A Generation Task Specification Condition which determines whether the value entered is even.
    /// </summary>
    /// <remarks><see cref="ComponentTaskReleaseCondition"/> are evaluated at the time of releasing component-sets.</remarks>
    [ComponentTaskCondition("Is Even",
                            "Passes if the specified value is even.",
                            "embedded://DriveWorks.Sdk.Examples.CSharp.Puzzle-16x16.png",
                            "SDK-Starter-Examples Plugin")]
    public class MyGenerationTaskSpecificationCondition : ComponentTaskReleaseCondition
    {
        private readonly FlowProperty<double> mValue;

        public MyGenerationTaskSpecificationCondition()
        {
            mValue = this.Properties.RegisterDoubleProperty("Value",
                                                            "The value to check for must be an even number for the condition to pass.");
        }

        protected override bool Evaluate(SpecificationContext specificationContext)
        {
            // If a number is evenly divisible by 2 with no remainder, then it is even.
            var isEven = (mValue.Value % 2) == 0;

            return isEven;
        }
    }
}